sys/kern/src/fail.rs

```diff
@@ -42,8 +42,11 @@ static mut KERNEL_EPITAPH: [u8; EPITAPH_LEN] = [0; EPITAPH_LEN];
 #[cfg(not(feature = "nano"))]
 fn begin_epitaph() -> &'static mut [u8; EPITAPH_LEN] {
     // We'd love to use an AtomicBool here but we gotta support ARMv6M.
-    let previous_fail =
-        core::mem::replace(unsafe { &mut KERNEL_HAS_FAILED }, true);
+    // This could probably become SyncUnsafeCell in a future where it exists.
+    let previous_fail = core::mem::replace(
+        unsafe { &mut *core::ptr::addr_of_mut!(KERNEL_HAS_FAILED) },
+        true,
+    );
```

Comment:
Alternatively, as per the above comment re: `AtomicBool`, could we use `armv6m-atomic-hack` here instead? Or is there a reason not to?

Reply:
That doesn't work in the kernel. It could probably be made to work in the kernel, but I haven't really looked into it yet.

Reply:
Are you sure you don't want to use `ptr::replace` here? Reborrowing into `&mut` after you've used `addr_of_mut!` is odd.

Reply:
Oh I see, the point of this function is to return `&'static mut`. So I suppose some friction with the lint is unavoidable.

Reply:
Hi @saethlin! I think you're right actually, because the two blocks are operating on two different statics. I have replaced the replace with `core::ptr::replace`, which it appears I didn't know existed when I wrote this.